tests/test_date.py

```diff
@@ -132,76 +204,171 @@ def test_periods(self, low, high, period, expected_results):
-        low = datetime(2014, 6, 15)
-        period = datetime(2014, 6, 25)
-
-        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(invalid_period)):
-            date.get_intersecting_periods(low, period, period=invalid_period).next()
+        self.given_period_low(datetime(2014, 6, 15))
+        self.given_period(datetime(2014, 6, 25))
+        self.given_invalid_period(invalid_period)
+        self.then_should_reject_easily_mistaken_dateutil_arguments()
 
     @parameterized.expand([
-        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14)),
-        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25)),
+        param(low=datetime(2014, 4, 15), high=datetime(2014, 4, 14), period='month'),
+        param(low=datetime(2014, 4, 25), high=datetime(2014, 4, 25), period='month'),
     ])
-    def test_empty_period(self, low, high):
-        results = list(date.get_intersecting_periods(low, high, period='month'))
-        self.assertEquals([], results)
+    def test_empty_period(self, low, high, period):
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_period(period)
+        self.when_empty_period()
+
+    def given_expected_results(self, expected_results):
+        self.expected_results = expected_results
+
+    def given_period_low(self, low):
+        self.period_low = low
+
+    def given_invalid_period(self, invalid_period):
+        self.invalid_period = invalid_period
+
+    def given_period_high(self, high):
+        self.period_high = high
+
+    def given_period(self, period):
+        self.period = period
 
+    def when_date_intersecting_generated(self, dates):
+        self.intersected_dates = dates
 
-class ParseDateWithFormats(unittest.TestCase):
+    def then_date_arguments_and_date_range_length_should_equal(self):
+        self.assertEquals(1, len(self.intersected_dates))
+
+    def then_date_period_and_intersected_dates_should_equal(self):
+        self.assertEquals(self.period_low, self.intersected_dates[0])
+
+    def when_should_one_date_for_each_month(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def when_empty_period(self):
+        self.intersected_dates = list(date.get_intersecting_periods(
+            self.period_low, self.period_high, period=self.period))
+
+    def then_should_one_date_for_each_month(self):
+        self.assertEquals(self.expected_results, self.intersected_dates)
+
+    def then_empty_period(self):
+        self.assertEquals([], self.intersected_dates)
+
+    def then_should_reject_easily_mistaken_dateutil_arguments(self):
+        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(self.invalid_period)):
+            date.get_intersecting_periods(self.period_low, self.period, period=self.invalid_period).next()
```

Comment:
I am not sure on splitting this in 2 as you suggested. If I use ```date.get_intersecting_periods(self.period_low, self.period, period=self.invalid_period).next()``` as a action then it will fail since the period will be invalid.

Give me an idea of what you had in mind.